diff.go

```diff
@@ -0,0 +1,65 @@
+package manifestival
+
+import (
+	"encoding/json"
+
+	"github.com/manifestival/manifestival/patch"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type JSONMergePatch map[string]interface{}
+
+// Diff returns a list of JSON Merge Patches [RFC 7386] that represent
+// the changes that will occur when the manifest is applied
+func (m Manifest) Diff() ([]JSONMergePatch, error) {
+	diffs, err := m.diff()
+	if err != nil {
+		return nil, err
+	}
+	result := make([]JSONMergePatch, len(diffs))
+	for i, bytes := range diffs {
+		if err := json.Unmarshal(bytes, &result[i]); err != nil {
+			return nil, err
+		}
+	}
+	return result, nil
+}
+
+// diff loads the resources in the manifest and computes their difference
+func (m Manifest) diff() (result [][]byte, err error) {
```

Comment:
Declaring "result" like this has two problems:

1. You don't get a chance to pre-allocate elements for the outer slice (which is max `len(m.resources)` and avoids some copying.
2. It's possible to return partial results, because you fill `result` in as you go, but sometimes you early-exit on error.

Reply:
Done, thanks